### DISCUSSION TITLE
HttpError: Not Found

### DISCUSSION THREAD

**Discussion by heyteresagreen** [ObsidianPublisher/obsidian-github-publisher#49] [Q&A — Plugin]
Hello! I'm so excited to find this plugin, but when I try to run it I see the error 
`HttpError: Not Found`. 

I've tried changing the 'Folder behaviour' settings and using a different repository, but still no luck. I'm using a Digital Garden Jekyll template and want to upload files to the _notes folder. Am I looking the right place to fix this?

I also have "Notice every error" turned on but not sure where it's logging to.

**Comment by Mara-Li** [comment]
This error seems to happend when the github repository is not found.
What's your settings ?

**Reply by heyteresagreen** [reply]
Thanks for the quick reply! Here's my settings:

Repo name: heyteresagreen.github.io (I've also tried this with heyteresagreentest in case the . is an issue)
Username: heyteresagreen
I've generated a token as described, Github says it's never been used

Folder behaviour: Obsidian Path (also tried Fixed)
Default folder: _notes (also tried blank)
Path removing: blank (also tried heyteresagreen.github.io )

**Comment by Mara-Li** [comment]
You need to just use the repo pure name! Not the links.
For example, the repo name here is `obsidian-github-publisher`

**Reply by heyteresagreen** [reply]
That is what I named my repo [heyteresagreen.github.io](https://github.com/heyteresagreen/heyteresagreen.github.io) Does that sound right? Sorry I'm new to Github.

I did try it with another repo with no . in the name, same issue

**Reply by Mara-Li** [reply]
I don't think it's a good idea to name it like this. I never thinked that it's possible, because, well, github pages have strange behavior.
I think the plugin I use don't found it :/ Could you rename it ?

**Reply by heyteresagreen** [reply]
I've also tried it with [heyteresagreentest](https://github.com/heyteresagreen/heyteresagreentest) in case the name is the issue, I'm still getting the same error though :(

**Reply by Mara-Li** [reply]
You need to rename your repository.

**Reply by heyteresagreen** [reply]
It's a second repository with a different name - I just wanted to test it before changing the name of my first one. It's still not working for me though.

**Reply by Mara-Li** [reply]
So, you set a repository with : 
- Repo name : `heyteresagreentest`
- username: `heyteresagreentest`
- A good token 
- the good branch (aka : `master`)
?

**Reply by heyteresagreen** [reply]
Oh! I renamed my Obsidian vault (which was also heyteresagreen.github.io to just heyteresagreen, and now I'm getting a different error:

`HttpError: Bad Credentials`

I feel like that's closer?

**Reply by heyteresagreen** [reply]
Yes, except:

Username: heyteresagreen

It should be my Github username right?

**Reply by Mara-Li** [reply]
Oh, maybe a obsidian vault with `.` in it can crash the function to create branch.

**Reply by Mara-Li** [reply]
> Yes, except:
> 
> Username: heyteresagreen
> 
> It should be my Github username right?

Yes

**Reply by tobiasvl** [reply]
Just FYI, I use Github Publisher with my GitHub Pages repo, which is called [tobiasvl.github.io](https://github.com/tobiasvl/tobiasvl.github.io), and it works fine. So I'm not sure that's the issue.

**Reply by heyteresagreen** [reply]
Thanks @tobiasvl, maybe that means I won't have to rename it!

**Comment by heyteresagreen** [comment]
So it seems that having "." in the name of the Obsidian vault causes the error: 
`HttpError: Not Found`

**Comment by heyteresagreen** [comment]
Now that I've renamed my vault and am testing with a different Github repo which also has no "." in the name, I'm getting a different error:

`HttpError: Bad Credentials`

I've tried regenerating my token and getting a new token, but still getting this error. Would it be a token related issue, or should I be looking somewhere else?

**Reply by tobiasvl** [reply]
> I also have "Notice every error" turned on but not sure where it's logging to.

Try opening the Developer Tools by pressing <kbd>Ctrl</kbd>+<kbd>Shift</kbd>+<kbd>i</kbd> (Windows, Mac probably uses Cmd instead) and clicking "Console", you should see the errors there.

**Reply by heyteresagreen** [reply]
Thank you! I was able to open it up and take a look at the errors

**Comment by heyteresagreen** [comment]
I just kept moving things around and now it seems to be working. I'm not sure exactly what did it, as I made a bunch of different repositories and vaults with different names. But I'm happy it's working now!

**Comment by Mara-Li** [comment]
Issue probably linked to #31
